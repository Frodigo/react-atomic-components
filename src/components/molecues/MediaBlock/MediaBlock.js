import React from 'react';
import PropTypes from 'prop-types';

import Heading from '../../atoms/Text/Heading';
import Paragraph from '../../atoms/Text/Paragraph';

import classes from './MediaBlock.module.css';

const MediaBlock = (props) => {
    const {
        imageUrl,
        linkUrl,
        title,
        text,
        imageWidth,
        imageHeight,
        ...rest
    } = props;

    const mediaBlockImage = imageUrl ? (
        <img
            src={imageUrl}
            alt={title}
            width={imageWidth}
            height={imageHeight}
        />
    ) : null;

    return (
        <div className={classes.mediaBlock} {...rest}>
            <div className={classes.mediaBlockInner}>
                <figure className={classes.image}>{mediaBlockImage}</figure>
                <section className={classes.content}>
                    <Heading type="h5">{title}</Heading>
                    <Paragraph>{text}</Paragraph>
                </section>
            </div>
        </div>
    );
};

MediaBlock.propTypes = {
    imageUrl: PropTypes.string,
    linkUrl: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

export default MediaBlock;
