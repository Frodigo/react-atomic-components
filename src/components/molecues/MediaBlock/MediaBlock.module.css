.mediaBlock {
    padding: 0 var(--spacing-md);
    margin: var(--spacing-md) auto
}

.mediaBlockInner {
    color: var(--color-text);
    text-decoration: none;
}

.content {
    padding: var(--spacing-md) 0;
}

.content h5 {
    margin-top: 0;
    margin-bottom: var(--spacing-md);
}

.content p {
    margin: 0;
    font-size: 14px;
    font-weight: 400;
}

.image {
    display: block;
    height: auto;
    width: 100%;
}

.image svg {
    height: auto;
    width: 100%;
}

@media (min-width: 768px) {
    .mediaBlock {
        margin: var(--spacing-xxl) auto
    }

    .mediaBlockInner {
        align-items: center;
        display: flex;
    }

    .image {
        flex: 1 1 30%;
    }

    .content {
        padding-left: 50px;
        flex: 1 1 70%;
    }
}
