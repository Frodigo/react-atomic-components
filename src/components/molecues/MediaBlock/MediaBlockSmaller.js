import React from 'react';
import PropTypes from 'prop-types';

import Heading from '../../atoms/Text/Heading';

import classes from './MediaBlockSmaller.module.css';

const MediaBlockSmaller = (props) => {
    const {
        imageWidth,
        imageHeight,
        imageUrl,
        subtitle,
        title,
        text,
        roundedImage,
        ...rest
    } = props;
    const mayDisplaySubtitle = subtitle ? (
        <span className={classes.subtitle}>{subtitle}</span>
    ) : null;

    const mediaBlockImage =
        imageUrl && imageUrl ? (
            <figure
                className={
                    roundedImage ? classes.roundedFigure : classes.figure
                }
            >
                <img
                    src={imageUrl}
                    alt={title}
                    width={imageWidth}
                    height={imageHeight}
                />
            </figure>
        ) : null;

    return (
        <div className={classes.mediaBlockSmaller} {...rest}>
            {mediaBlockImage}
            <section className={classes.content}>
                {mayDisplaySubtitle}
                <Heading type="h3">{title}</Heading>
                {text}
            </section>
        </div>
    );
};

MediaBlockSmaller.propTypes = {
    imageHeight: PropTypes.number,
    imageUrl: PropTypes.string,
    imageWidth: PropTypes.number,
    roundedImage: PropTypes.bool,
    subtitle: PropTypes.string,
    text: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

export default MediaBlockSmaller;
