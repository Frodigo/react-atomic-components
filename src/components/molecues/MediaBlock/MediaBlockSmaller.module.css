.mediaBlockSmaller {
    border-bottom: solid 2px var(--color-secondary);
    margin: 40px 0;
    padding: 0 var(--spacing-md) var(--spacing-xxl) var(--spacing-md);
    text-align: center;
}

.mediaBlockSmaller h3 {
    margin-top: 0;
    margin-bottom: 5px;
}

.mediaBlockSmaller p {
    font-size: 18px;
}

.subtitle {
    display: block;
    font-size: 16px;
    font-weight: 700;
    margin-top: 12px;
    line-height: 1.75;
}

.roundedFigure {
    border: solid 1px #979797;
    border-radius: 50%;
    height: 270px;
    margin: 0 auto 30px auto;
    overflow: hidden;
    width: 270px;
}

.roundedFigure,
.figure {
    flex-shrink: 0;
}

@media (min-width: 1024px) {
    .roundedFigure {
        margin-bottom: 0;
    }

    .mediaBlockSmaller {
        align-items: center;
        display: flex;
        text-align: left;
    }

    .content {
        padding-left: var(--spacing-md);
    }
}

@media (min-width: 1300px) {
    .content {
        padding-left: var(--spacing-xxl);
    }
}
