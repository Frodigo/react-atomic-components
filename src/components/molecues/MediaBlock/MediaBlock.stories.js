import React from 'react';

import { default as MediaBlockComponent } from './MediaBlock';

export default {
    title: 'Molecues/MediaBlock/default',
    component: MediaBlockComponent
};

const Template = (args) => <MediaBlockComponent {...args} />;

export const MediaBlock = Template.bind({});
MediaBlock.args = {
    imageUrl:
        'https://via.placeholder.com/400x240',
    imageWidth: 400,
    imageHeight: 240,
    linkUrl: '#',
    text:
        'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. ',
    title: 'Lorem ipsum dolor sit'
};
