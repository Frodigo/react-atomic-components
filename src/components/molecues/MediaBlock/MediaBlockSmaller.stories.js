import React from 'react';

import { default as MediaBlockSmallerComponent } from './MediaBlockSmaller';

export default {
    title: 'Molecues/MediaBlock/smaller',
    component: MediaBlockSmallerComponent
};

const Template = (args) => <MediaBlockSmallerComponent {...args} />;

export const MediaBlockSmaller = Template.bind({});
MediaBlockSmaller.args = {
    imageUrl: 'https://via.placeholder.com/290x240',
    imageWidth: 290,
    imageHeight: 240,
    subtitle: '05/2016 - 04/2020',
    text: 'dolorm sit amet',
    title: 'Lorem ipsum'
};
