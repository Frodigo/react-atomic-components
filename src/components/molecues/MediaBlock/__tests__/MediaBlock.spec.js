import React from 'react';
import { render } from '@testing-library/react';

import MediaBlock from '../MediaBlock';

describe('MediaBlock should be rendered', () => {
    const props = {
        imageUrl: 'https://via.placeholder.com/290x240',
        linkUrl: '#',
        text:
            'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. ',
        title: 'Lorem ipsum dolor sit'
    };

    test('MediaBlock renders correctly', () => {
        const { container } = render(<MediaBlock {...props} />);

        expect(container).toMatchSnapshot();
    });
});
