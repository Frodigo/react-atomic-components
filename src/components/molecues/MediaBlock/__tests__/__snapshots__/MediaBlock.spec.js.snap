// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaBlock should be rendered MediaBlock renders correctly 1`] = `
<div>
  <div
    class="mediaBlock"
  >
    <div
      class="mediaBlockInner"
    >
      <figure
        class="image"
      >
        <img
          alt="Lorem ipsum dolor sit"
          src="https://via.placeholder.com/290x240"
        />
      </figure>
      <section
        class="content"
      >
        <h5
          class=""
        >
          Lorem ipsum dolor sit
        </h5>
        <p
          class="paraghaph"
        >
          Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. 
        </p>
      </section>
    </div>
  </div>
</div>
`;
